.github/assert-deployed-bytecode.js

```diff
@@ -0,0 +1,104 @@
+const { AssertionError } = require('chai')
+const chalk = require('chalk')
+const { web3 } = require('hardhat')
+const { readJSON } = require('../scripts/helpers/fs')
+const { APPS_TO_NAMES, CONTRACTS_TO_NAMES, IGNORE_METADATA_CONTRACTS } = require('./deployed-bytecode-consts')
+
+function compareSymbolWise(first, second, ignoreMetadata) {
+  let compareEnd = first.length
+  if (ignoreMetadata) {
+    compareEnd = first.indexOf('a264')
+  }
+  for (var i = 0; i < compareEnd; i++) {
+    if (first[i] != second[i] && first[i] != '0' && second[i] != '0') {
+      return false
+    }
+  }
+  return true
+}
+
+async function assertByteCode(address, artifactName, deployTx) {
+  const artifact = await artifacts.readArtifact(artifactName)
+  let bytecodeFromArtifact = artifact.deployedBytecode.toLowerCase()
+  const bytecodeFromRpc = (await web3.eth.getCode(address)).toLowerCase()
+  const ignoreMetadata = IGNORE_METADATA_CONTRACTS.includes(artifactName)
+  if (bytecodeFromRpc === bytecodeFromArtifact) {
+    console.log(chalk.green(`Compiled bytecode for ${chalk.yellow(address)}(${artifactName}) MATCHES deployed bytecode!`))
+  } else if (
+    bytecodeFromRpc.length == bytecodeFromArtifact.length &&
+    compareSymbolWise(bytecodeFromArtifact, bytecodeFromRpc, ignoreMetadata)
+  ) {
+    console.log(chalk.hex('#FFA500')(`Compiled bytecode for ${chalk.yellow(address)}(${artifactName}) is SIMILAR to deployed bytecode!`))
+    if (deployTx) {
+      await assertByteCodeByDeployTx(address, deployTx, artifact)
+    } else if (!ignoreMetadata) {
```

Comment:
It might be a bit incorrect since if even we ignore IPFS metadata, we don't ignore the immutable vars.

Reply:
Can't get your idea here. This is a workaround for the wstETH. Yet, we can also add `ignoreMetadata` to `assertByteCodeByDeployTx`

Reply:
Yep. Already done this way